iree/tools/iree-run-module-main.cc

```diff
@@ -160,6 +160,15 @@ iree_status_t Run() {
 
 extern "C" int main(int argc, char** argv) {
   iree_flags_parse_checked(IREE_FLAGS_PARSE_MODE_DEFAULT, &argc, &argv);
+  if (argc > 1) {
+    // Avoid iree-run-module spinning endlessly on stdin if the user uses single
+    // dashes for flags.
+    std::cout << "Error: More than one positional argument (single dash '-') "
+                 "given for initiating stdin input. Perhaps you meant to use "
+                 "'--' instead?"
+              << std::endl;
```

Comment:
I don't think the error message here is quite right. iree-run-module doesn't accept *any* positional arguments. argc will include the name of the program itself, which is why this is `> 1`, but I wouldn't count that as a positional argument. Also, the reason that it's looking for stdin is because that's the default for `module_file`. If you pass with a single dash then it gets interpreted as a positional argument, so `module_file` remains unset and defaults to stdin. Also, don't use std::endl. It's basically never the right thing to use


```suggestion
    std::cout << "Error: unexpected positional argument (expected none)."
                 " Did you use pass a flag with a single dash ('-')?"
                 " Use '--' instead.\n";
```